spec/javascript/controllers/talk-to-us_controller_spec.js

```diff
@@ -61,22 +64,40 @@ describe('TalkToUsController', () => {
 
       setBody(true);
       setCurrentTime('2021-01-01 10:00');
-      registerController();
     });
 
-    it('opens the Zendesk Chat widget when clicking the button', () => {
+    it('appends the Zendesk snippet, opens the chat window and shows a loading message when clicking the button', () => {
       const button = document.querySelector('a');
       button.click();
-
+      expect(document.querySelector('#ze-snippet')).not.toBeNull();
+      expect(getButtonText()).toEqual("Starting chat...");
+      jest.runOnlyPendingTimers(); // Timer for script loading,
+      jest.runOnlyPendingTimers(); // Timer to wait for chat window to open.
       expect(chatShowSpy).toBeCalledWith();
+      expect(getButtonText()).toEqual("Chat Online");
+    });
+
+    describe('when clicking the chat button twice', () => {
+      it('only appends the Zendesk snippet once and does not show the loading message for the second click', () => {
+        const button = document.querySelector('a');
+        button.click();
+        expect(button.textContent).toEqual("Starting chat...");
+        jest.runOnlyPendingTimers(); // Timer for script loading,
+        jest.runOnlyPendingTimers(); // Timer to wait for chat window to open.
+        expect(chatShowSpy).toBeCalledWith();
```

Comment:
Should this `toBeCalledWith` have a value?

Reply:
It works without, but yeah I'll change it to `toHaveBeenCalled` as that reads much nicer 👍